### ISSUE TITLE
S3 bucket grantee

### ISSUE THREAD

**Issue by hashibot** [hashicorp/terraform-provider-aws#247]
_This issue was originally opened by @pporada-gl as hashicorp/terraform#8583. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

Terraform v0.7.2
### Affected Resource(s)

Please list the resources as a list, for example:
- aws_s3_bucket
### Expected Behavior

I should be able to set a grantee on a bucket. Currently I have to use a null_resource and a local_exec to run

```
    $ aws s3api put-bucket-acl --bucket mybucket --grant-write id=XXXXXXXXXXXXXXXXX
```
### Actual Behavior

I can't set a grantee on a bucket via specifically the aws_s3_bucket resource though it is supported in the Go SDK.
### Important Factoids

The AWS Go SDK supports grants on S3 buckets.
https://github.com/aws/aws-sdk-go/blob/master/models/apis/s3/2006-03-01/api-2.json#L2331

**Comment by rokka-n** [comment]
It would be nice to have this supported.

**Comment by jingyimei** [comment]
+1

**Comment by kmacoskey** [comment]
+1

**Comment by divyabhargov** [comment]
+1

**Comment by CpuID** [comment]
+1

**Comment by segersniels** [comment]
+1

**Comment by allandrick** [comment]
+1

**Comment by jonathanhle** [comment]
+1

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/144

**Comment by FearlessHyena** [comment]
Would be great if this was supported

**Comment by ewbankkit** [comment]
@pporada-gl @FearlessHyena Does the new [`aws_s3_bucket.grant`](https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#using-acl-policy-grants) argument added via https://github.com/terraform-providers/terraform-provider-aws/issues/989, https://github.com/terraform-providers/terraform-provider-aws/pull/3728 and available in [version 2.52.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/v2.52.0/CHANGELOG.md) address your use case?

**Comment by FearlessHyena** [comment]
@ewbankkit Thanks for the reference. I didn't realise they added to the AWS provider recently. Let me try it out and see if it addresses my use case

**Comment by anGie44** [comment]
Hi all, closing this is it's also available via the new [`aws_s3_bucket_acl.access_control_policy` parameter](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_acl#access_control_policy) available since `v3.75.0` of the Terraform AWS Provider.
